### PR TITLE
Add an in-cluster example for error injection

### PR DESCRIPTION
**Pull request by yuanchen8911**
#### What type of PR is this?

/kind feature
#### What this PR does / why we need it:

This PR adds an example showing how to inject an error to an initContainer to simulate failure in a Kubernetes cluster.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: site/content/en/docs/user/examples/injectfailure/virtual-gpu-node.yaml [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    intel.com/mlnx_sriov_rdma5: "1"
+    intel.com/mlnx_sriov_rdma6: "1"
+    intel.com/mlnx_sriov_rdma7: "1"
+    intel.com/mlnx_sriov_rdma8: "1"
+    memory: 1Ti
+    nvidia.com/gpu: "8"
+    pods: "110"
+  conditions:
+  - lastHeartbeatTime: "2024-04-26T16:14:04Z"
+    lastTransitionTime: "2024-04-26T16:14:04Z"
```
Comment (wzshiming) [inline site/content/en/docs/user/examples/injectfailure/virtual-gpu-node.yaml:64]:
```suggestion
  - lastHeartbeatTime: null
    lastTransitionTime: null
```

File: site/content/en/docs/user/examples/injectfailure/failed-pod.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  initContainers:
+  - image: nccl
+    name: nccl-test
+  containers:
+  - image: llm
+    name: training-job
+    resources:
+      requests:
+        cpu: 100m
+        memory: 100Mi
```
Comment (wzshiming) [inline site/content/en/docs/user/examples/injectfailure/failed-pod.yaml:39]:
```suggestion
        cpu: 100m
        memory: 100Mi
        nvidia.com/gpu: '1'
```

File: site/content/en/docs/user/examples/injectfailure/init-container-error-injection.demo [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# Demo: inject fault to initContainer in an existing minikube cluster
+kubectl config use-context minikube
+
+kubectl get nodes
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+WORK_DIR=$(cd ${SCRIPT_DIR}/../site/content/en/docs/user/examples/failureinjection/; pwd -P)
```
Comment (wzshiming) [inline site/content/en/docs/user/examples/injectfailure/init-container-error-injection.demo:6]: #1066
I think it should be a builtin variable

**Review by wzshiming — COMMENTED** [review]
Does this need to be supplemented with docs for a more complete example?

File: site/content/en/docs/user/examples/injectfailure/init-container-error-injection.demo (hunk 3 above)
Comment (yuanchen8911) [reply site/content/en/docs/user/examples/injectfailure/init-container-error-injection.demo:6]: OK, I'll make a change after your PR is merged.

**Comment by yuanchen8911** [comment]
Applied the changes and regenerated the svg file.

**Comment by yuanchen8911** [comment]
> example?

I don't think we need another one since it will be similar to the init container one.

**Comment by yuanchen8911** [comment]
> Does this need to be supplemented with docs for a more complete example?

Add a MD description file?

**Pull request merged**